src/main/java/io/confluent/connect/jdbc/sink/metadata/FieldsMetadata.java

```diff
@@ -52,26 +52,27 @@ public static FieldsMetadata extract(
       final String tableName,
       final JdbcSinkConfig.PrimaryKeyMode pkMode,
       final List<String> configuredPkFields,
+      final Set<String> fieldsWhitelist,
       final SchemaPair schemaPair
   ) {
-    return extract(tableName, pkMode, configuredPkFields, schemaPair.keySchema, schemaPair.valueSchema);
+    return extract(tableName, pkMode, configuredPkFields, fieldsWhitelist, schemaPair.keySchema, schemaPair.valueSchema);
   }
 
   public static FieldsMetadata extract(
```

Comment:
The logic in here is extremely hairy since tnhere is interplay of `pk.mode`, `pk.fields` and now `fields.whitelist` against what we actually find in the key & value schema of a record.